router/h2/src/main/scala/io/buoyant/router/h2/StreamStatsFilter.scala

```diff
@@ -21,40 +21,56 @@ object StreamStatsFilter {
 class StreamStatsFilter(statsReceiver: StatsReceiver)
   extends SimpleFilter[Request, Response] {
 
-  private[this] val reqStreamTimeMs = statsReceiver.stat("stream", "request", "duration_ms")
-  private[this] val rspStreamTimeMs = statsReceiver.stat("stream", "response", "duration_ms")
-  private[this] val streamTimeMs = statsReceiver.stat("stream", "duration_ms")
-  private[this] val reqStreamFailures = statsReceiver.counter("stream", "request", "failures")
-  private[this] val rspStreamFailures = statsReceiver.counter("stream", "response", "failures")
-  private[this] val streamFailures = statsReceiver.counter("stream", "failures")
+  class StreamStats(
+    stats: StatsReceiver,
+    durationName: Option[String] = None,
+    successName: Option[String] = None
+  )(scopes: String*) {
 
-  private[this] val latencyMs = statsReceiver.stat("latency_ms")
+    private[this] val (durationMs, successes, failures) = {
+      val scope = scopes.foldLeft(statsReceiver) { (scope, name) => scope.scope(name) }
+      (scope.stat(s"${durationName.getOrElse("stream_duration")}_ms"),
+       scope.counter(s"${successName.getOrElse("stream_")}successes"),
+       scope.counter(s"${successName.getOrElse("stream_")}failures"))
+    }
 
-  override def apply(req: Request, service: Service[Request, Response]): Future[Response] = {
-    val reqT = Stopwatch.start()
-    req.stream.onEnd.respond {
-      case Return(_) => reqStreamTimeMs.add(reqT().inMillis)
-      case Throw(_) => reqStreamFailures.incr()
+    @inline def apply(startT: Stopwatch.Elapsed)(result: Try[_]): Unit = {
+      durationMs.add(startT().inMillis)
+      result match {
+        case Return(_) => successes.incr()
+        case Throw(_) => failures.incr()
+      }
     }
+  }
 
-    val rspF = service(req)
+  //   total number of requests received
+  private[this] val reqCount = statsReceiver.counter("requests")
 
-    rspF.onSuccess { rsp =>
-      latencyMs.add(reqT().inMillis)
+  private[this] val reqStreamStats = new StreamStats(statsReceiver)("stream", "request")
+  private[this] val rspStreamStats = new StreamStats(statsReceiver)("stream", "response")
+  private[this] val totalStreamStats = new StreamStats(statsReceiver, Some("total_latency"))("stream")
+  private[this] val rspFutureStats = new StreamStats(statsReceiver, Some("response_latency"), Some(""))("response")
```

Comment:
I'd consider not scoping this under "response" and calling the duration "request_latency".  It's a less accurate name but it's more consistent with other protocols and will make play more nicely with the admin dashboard.